studio/data/database-triggers/database-trigger-update-transaction-mutation.ts

```diff
@@ -0,0 +1,66 @@
+import { useMutation, UseMutationOptions, useQueryClient } from '@tanstack/react-query'
+import { executeSql } from 'data/sql/execute-sql-query'
+import { databaseTriggerKeys } from './keys'
+import { quoteLiteral } from 'lib/pg-format'
+
+// [Joshen] Writing this query within FE as the PATCH endpoint from pg-meta only supports updating
+// trigger name and enabled mode. So we'll delete and create the trigger, within a single transaction
+// Copying the SQL from https://github.com/supabase/postgres-meta/blob/master/src/lib/PostgresMetaTriggers.ts
+
+export type DatabaseTriggerUpdateVariables = {
+  projectRef: string
+  connectionString?: string
+  originalTrigger: any
+  updatedTrigger: any
+}
+
+export function getDatabaseTriggerUpdateSQL({
+  originalTrigger,
+  updatedTrigger,
+}: Pick<DatabaseTriggerUpdateVariables, 'originalTrigger' | 'updatedTrigger'>) {
+  const { name, activation, events, schema, table, function_schema, function_name, function_args } =
+    updatedTrigger
+  return `
```

Comment:
Syntax highlighting for VS Code

```suggestion
  return /* SQL */ `
```